File: TypeScript/app/handler/SulfurasHandler.ts
import { Item } from '../gilded-rose';
import { ItemHandler } from './ItemHandler';

export class SulfurasHandler implements ItemHandler {
  handles(item: Item): boolean {
    return item.name === 'Sulfuras, Hand of Ragnaros';
  }

  updateItemQuality(item: Item): void {
    // Sulfuras does not change in quality or sellIn
  }
}

File: Java/src/main/java/com/gildedrose/handler/SulfurasHandler.java
package com.gildedrose.handler;

import com.gildedrose.Item;

public class SulfurasHandler implements ItemHandler {
    @Override
    public boolean handles(String itemName) {
        return "Sulfuras, Hand of Ragnaros".equals(itemName);
    }

    @Override
    public void updateItem(Item item) {
        // Legendary item, no changes to quality or sellIn
    }
}
